### ISSUE TITLE
[Feature] support $var() for textops in "re" key

### ISSUE THREAD

**Issue by netaskd** [OpenSIPS/opensips#1411]
Please add $var() support into textops module in "re" key. Now it supports only static values as "re"
E.g.:
```
$var(bad_ua_re)="scanner|sip-scan";
if ( search_body("$var(bad_ua_re)") || search("$var(bad_ua_re)")  ) {
.....
}
```

**Comment by netaskd** [comment]
Hi @fgast 
It doesn't works. My test failed.
I do:
```
[route] {
...
  $var(bad_ua_re)="scanner|sip-scan";
$log_level=6;
  if ( search_body("$var(bad_ua_re)") || search("$var(bad_ua_re)") ) {
    xlog("is matched\n");
  } else {
    xlog("is not matched\n");
  }
$log_level=null;
...
```
set body as:
```
...
      Max-Forwards: 70
      Subject: sip-scan
      Content-Type: application/sdp
      Content-Length: [len]

      v=0
      o=sip-scan 53655765 2353687637 IN IP[local_ip_type] [local_ip]
      s=-
      c=IN IP[media_ip_type] [media_ip]
      t=0 0
      m=audio [media_port] RTP/AVP 9 101
...
```
and its return
```
Jul 16 18:50:55 [4160] DBG:core:parse_headers: flags=ffffffffffffffff
Jul 16 18:50:55 [4160] is not matched
```

**Comment by fgast** [comment]
Hi @netaskd 
Your configuration gives me this result: 

`Jul 24 16:04:50 [7994] DBG:core:get_hdr_field: cseq <CSeq>: <1> <INVITE>
Jul 24 16:04:50 [7994] DBG:maxfwd:is_maxfwd_present: value = 70
Jul 24 16:04:50 [7994] DBG:core:parse_headers: flags=ffffffffffffffff
Jul 24 16:04:50 [7994] DBG:core:get_hdr_field: content_length=142
Jul 24 16:04:50 [7994] DBG:core:get_hdr_field: found end of header
Jul 24 16:04:50 [7994] is matched
Jul 24 16:04:50 [7994] DBG:tm:t_newtran: transaction on entrance=0xffffffffffffffff
Jul 24 16:04:50 [7994] DBG:core:parse_headers: flags=ffffffffffffffff`

Maybe you can send me your sipp scenario and a bit more of the opensips log so i can look at it again.

**Comment by netaskd** [comment]
Hi @fgast 
maybe we using different version?
I use: 
```
$ opensips -V
version: opensips 2.4.1 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: e881143
main.c compiled on  with gcc 5.3.1
```

sipp scenario like this:
https://pastebin.com/qqcY5aaP

opensips log here:
https://pastebin.com/1KHkCwxR

but refers to textops module only this lines:
```
Aug  6 16:53:15 [1404] DBG:core:parse_headers: flags=ffffffffffffffff
Aug  6 16:53:15 [1404] is not matched
```
it doesn't provide more information in log_level=6

**Comment by fgast** [comment]
Hello @netaskd 
I also guess it is about your version.
Where did you get the sources from? Did you apply my patch? 
This stuff is not (yet) merged into opensips and i guess it will not go into 2.4.x as it is a new feature.

**Comment by netaskd** [comment]
Hi @fgast ,
I've applied the patch to commit 66114a6 (tag 2.4.2) and got Segfault:
Here a snippet of bt full:
```
Core was generated by `opensips -Ff /etc/opensips/opensips.cfg -w /tmp'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  __GI___regexec (preg=0x1, 
    string=0x5639ed14393c "Via: SIP/2.0/UDP 199.199.199.198:5060;branch=z9hG4bK7ed59071;rport\r\nMax-Forwards: 69\r\nFrom: \"AAAA\" <sip:AAAA@199.199.199.198>;tag=as0f3f632c\r\nTo: <sip:111@sip.example.net:5080;transport=udp>\r\nCon"..., nmatch=5, pmatch=0x7f304e4e84a8, eflags=0) at regexec.c:199
199	regexec.c: No such file or directory.
(gdb) bt full
#0  __GI___regexec (preg=0x1, 
    string=0x5639ed14393c "Via: SIP/2.0/UDP 199.199.199.198:5060;branch=z9hG4bK7ed59071;rport\r\nMax-Forwards: 69\r\nFrom: \"AAAA\" <sip:AAAA@199.199.199.198>;tag=as0f3f632c\r\nTo: <sip:111@sip.example.net:5080;transport=udp>\r\nCon"..., nmatch=5, pmatch=0x7f304e4e84a8, eflags=0) at regexec.c:199
        err = <optimized out>
        start = <optimized out>
        length = <optimized out>
        dfa = <optimized out>
        ignore1 = <optimized out>
        ignore2 = <optimized out>
        ignore3 = <optimized out>
        ignore = <optimized out>
#1  0x00005639ecdc6258 in subst_run ()
No symbol table info available.
#2  0x00007f304a3054e4 in ?? () from /usr/lib/x86_64-linux-gnu/opensips/modules/textops.so
No symbol table info available.
```

**Comment by netaskd** [comment]
up

**Comment by bogdan-iancu** [comment]
This is solved in 3.0 by the generic support of variables in the parameters of the script functions (exported by modules)
